src/draw_target.rs

```diff
@@ -247,6 +255,18 @@ pub(crate) enum Drawable<'a> {
 }
 
 impl<'a> Drawable<'a> {
+    pub(crate) fn last_line_count(&mut self) -> Option<&mut usize> {
```

Comment:
I think we should change all 3 `last_line_count()` methods to instead be more like `drop_lines(&mut self, num: usize)`, which would save on conditionals at call sites.